lib/teleterm/services/desktop/directorysharing_test.go

```diff
@@ -0,0 +1,176 @@
+// Teleport
+// Copyright (C) 2025 Gravitational, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package desktop
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/gravitational/trace"
+	"github.com/stretchr/testify/require"
+)
+
+func TestOpenSharedDirectory(t *testing.T) {
+	path := t.TempDir()
+	filePath := filepath.Join(path, testFilename)
+	err := os.WriteFile(filePath, []byte("test"), 0600)
+	require.NoError(t, err)
+	access := DirectoryAccess{}
+	err = access.Open(filePath)
+	require.True(t, trace.IsBadParameter(err), "%q is not a directory", filePath)
+}
+
+const (
+	testDirname         = "test_dir"
+	testFilename        = "test_file"
+	testSymlinkFilename = "test_symlink"
+)
+
+func setUpSharedDir(t *testing.T) (*DirectoryAccess, string) {
+	path := t.TempDir()
+	err := os.Mkdir(filepath.Join(path, testDirname), 0700)
+	require.NoError(t, err)
+	err = os.WriteFile(filepath.Join(path, testFilename), []byte("test"), 0600)
+	require.NoError(t, err)
+	err = os.Symlink(filepath.Join(path, testFilename), filepath.Join(path, testSymlinkFilename))
+	require.NoError(t, err)
+	access := DirectoryAccess{}
+	err = access.Open(path)
+	require.NoError(t, err)
+	return &access, path
+}
+
+func TestDirectoryAccessEscapingPaths(t *testing.T) {
+	outOfRootPath := filepath.Join(testDirname, "../..")
+	tests := []struct {
+		name string
+		call func(*DirectoryAccess) error
+	}{
+		{"Stat", func(a *DirectoryAccess) error { _, err := a.Stat(outOfRootPath); return err }},
+		{"ReadDir", func(a *DirectoryAccess) error { _, err := a.ReadDir(outOfRootPath); return err }},
+		{"Read", func(a *DirectoryAccess) error { _, err := a.Read(outOfRootPath, 0, 100); return err }},
+		{"Write", func(a *DirectoryAccess) error { _, err := a.Write(outOfRootPath, 0, []byte("test")); return err }},
+		{"Truncate", func(a *DirectoryAccess) error { err := a.Truncate(outOfRootPath, 100); return err }},
+		{"Create", func(a *DirectoryAccess) error { err := a.Create(outOfRootPath, FileTypeDir); return err }},
+		{"Delete", func(a *DirectoryAccess) error { err := a.Delete(outOfRootPath); return err }},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name+"_Escape", func(t *testing.T) {
+			t.Helper()
```

Comment:
```suggestion
```

You _don't_ want `t.Helper()` here, because the enclosing func is not a helper function, it is the test itself.